File: src/main/java/net/gamershub/kitpvp/commands/SpawnCommand.java
package net.gamershub.kitpvp.commands;

import net.gamershub.kitpvp.ExtendedPlayer;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class SpawnCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player p) {
            ExtendedPlayer extendedPlayer = ExtendedPlayer.from(p);

            if (extendedPlayer.getCombatCooldown() == 0) {
                p.teleport(new Location(Bukkit.getWorld("world"), 0.5, 100.5, -8.5, 0, 0));
                ExtendedPlayer.from(p).spawnPlayer();
            } else {
                p.sendMessage(Component.text("You are still in combat for " + extendedPlayer.getCombatCooldown() + "s!", NamedTextColor.RED));
            }
            return true;
        }
        return false;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/commands/SpawnCommand.kt
package io.github.chaosdave34.kitpvp.commands

import com.mojang.brigadier.Command
import io.github.chaosdave34.kitpvp.ExtendedPlayer
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.entity.Player

object SpawnCommand {
    fun register(commands: Commands) {
        commands.register(
            Commands.literal("spawn")
                .requires { it.sender is Player }
                .executes { spawn(it.source) }
                .build()
        )
    }

    private fun spawn(source: CommandSourceStack): Int {
        val player = source.sender as Player
        val extendedPlayer = ExtendedPlayer.from(player)

        if (extendedPlayer.combatCooldown == 0)
            extendedPlayer.spawn()
        else
            player.sendMessage(Component.text("You are still in combat for ${extendedPlayer.combatCooldown}s!", NamedTextColor.RED))

        return Command.SINGLE_SUCCESS
    }
}